pkgs/development/tools/misc/blackfire/default.nix

```diff
@@ -1,53 +1,40 @@
-{ stdenv
-, lib
-, fetchurl
-, dpkg
-, autoPatchelfHook
-, writeShellScript
-, curl
-, jq
-, common-updater-scripts
-}:
+{ stdenv, lib, fetchurl, dpkg, writeShellScript, curl, jq, autoPatchelfHook
+, common-updater-scripts }:
 
-stdenv.mkDerivation rec {
+let
   pname = "blackfire";
-  version = "2.5.2";
+  version = "2.8.0";
 
-  src = fetchurl {
-    url = "https://packages.blackfire.io/debian/pool/any/main/b/blackfire/blackfire_${version}_amd64.deb";
-    sha256 = "1RO5yabSNpIz5lWXngMOZ1S2vtnLEkXIj1ZoIinRrQ0=";
+  sources = {
+    "x86_64-linux" = fetchurl {
+      url =
+        "https://packages.blackfire.io/debian/pool/any/main/b/blackfire/blackfire_${version}_amd64.deb";
+      sha256 = "0bgd4hnpaxrqw0s0y2qiak8lbskfi2cqp147vj1kbhvm8834hwhg";
+    };
+    "i686-linux" = fetchurl {
+      url =
+        "https://packages.blackfire.io/debian/pool/any/main/b/blackfire/blackfire_${version}_i386.deb";
+      sha256 = "06lf642m4imz8xvwipflmvjy1ih7k8bx8jpay0xawvilh14pqz8f";
+    };
+    "aarch64-linux" = fetchurl {
+      url =
+        "https://packages.blackfire.io/debian/pool/any/main/b/blackfire/blackfire_${version}_arm64.deb";
+      sha256 = "0rddafjgdnj3na96x83paq5z14grj46v4iv38qbkvmdllrj26a0a";
+    };
+    "aarch64-darwin" = fetchurl {
+      url =
+        "https://packages.blackfire.io/blackfire/${version}/blackfire-darwin_aarch64.pkg.tar.gz";
+      sha256 =
+        "6168999d876c5bb77243421e29e0b5534d1921d2519f3b3d91e7904c4536a330";
+    };
+    "x86_64-darwin" = fetchurl {
+      url =
+        "https://packages.blackfire.io/blackfire/${version}/blackfire-darwin_amd64.pkg.tar.gz";
+      sha256 =
+        "391b0d239b11095bb8515cb60ee95f02d5862fcb509724081f314819967206b6";
+    };
   };
 
-  nativeBuildInputs = [
-    dpkg
-    autoPatchelfHook
-  ];
-
-  dontUnpack = true;
-
-  installPhase = ''
-    runHook preInstall
-
-    dpkg-deb -x $src $out
-    mv $out/usr/* $out
-    rmdir $out/usr
-
-    # Fix ExecStart path and replace deprecated directory creation method,
-    # use dynamic user.
-    substituteInPlace "$out/lib/systemd/system/blackfire-agent.service" \
-      --replace '/usr/' "$out/" \
-      --replace 'ExecStartPre=/bin/mkdir -p /var/run/blackfire' 'RuntimeDirectory=blackfire' \
-      --replace 'ExecStartPre=/bin/chown blackfire: /var/run/blackfire' "" \
-      --replace 'User=blackfire' 'DynamicUser=yes' \
-      --replace 'PermissionsStartOnly=true' ""
-
-    # Modernize socket path.
-    substituteInPlace "$out/etc/blackfire/agent" \
-      --replace '/var/run' '/run'
-
-    runHook postInstall
-  '';
-
   passthru = {
     updateScript = writeShellScript "update-${pname}" ''
       export PATH="${lib.makeBinPath [ curl jq common-updater-scripts ]}"
```

Comment:
In order to prevent this from going stale, could you also make the update script update all the platforms? You can take a look at https://github.com/NixOS/nixpkgs/blob/187764481b2bb6ce7ee825b497d4c91095622a17/pkgs/applications/editors/sublime/4/common.nix#L133-L138 for inspiration.